clar2wasm/src/wasm_generator.rs

```diff
@@ -247,6 +247,31 @@ impl WasmGenerator {
         })
     }
 
+    pub fn expand_memory_conditionally(&mut self) -> Result<(), GeneratorError> {
+        let memory = self
+            .module
+            .memories
+            .iter_mut()
+            .next()
+            .ok_or_else(|| GeneratorError::InternalError("No Memory found".to_owned()))?;
+
+        let total_module_space = memory.initial * 64 * 1024;
+        let new_memory_required = self.literal_memory_end;
+
+        if new_memory_required > total_module_space {
+            let pages_required = (new_memory_required - total_module_space) / (64 * 1024);
+            let remainder = (new_memory_required - total_module_space) % (64 * 1024);
+
+            if remainder > 0 {
+                memory.initial += pages_required + 1;
+            } else {
+                memory.initial += pages_required;
+            }
```

Comment:
Instead of having an `if` statement, we could have
```suggestion
        memory.initial += pages_required + (remainder > 0) as u32;
```